src/test/regress/sql/pg17.sql

```diff
@@ -1451,6 +1451,40 @@ SELECT * FROM sensor_readings ORDER BY 1;
 
 -- End of MERGE ... WHEN NOT MATCHED BY SOURCE tests
 
+-- Issue #7846: Test crash scenarios with MERGE on non-distributed and distributed tables
+-- Step 1: Connect to a worker node to verify shard visibility
+\c postgresql://postgres@localhost::worker_1_port/regression?application_name=psql
+SET search_path TO pg17;
+
+-- Step 2: Create and test a non-distributed table
+CREATE TABLE non_dist_table_12345 (id INTEGER);
+
+-- Test MERGE on the non-distributed table
+MERGE INTO non_dist_table_12345 AS target_0
+USING pg_catalog.pg_class AS ref_0
+ON target_0.id = ref_0.relpages
+WHEN NOT MATCHED THEN DO NOTHING;
+
+-- Step 3: Switch back to the coordinator for distributed table operations
+\c - - - :master_port
```

Comment:
Is it possible to connect to the coordinator like this:
`\c postgresql://postgres@localhost::master_port/regression?application_name=psql`

Reply:
done